libs/s25main/WineLoader.cpp

```diff
@@ -0,0 +1,164 @@
+// Copyright (C) 2024 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+///////////////////////////////////////////////////////////////////////////////
+
+#include "WineLoader.h"
+#include "Loader.h"
+
+// Wine Addon animation and images
+
+constexpr unsigned NOT_AVAILABLE = 0;
+constexpr unsigned EMPTY = 0;
+
+namespace wineaddon {
+std::map<std::pair<Nation, BuildingType>, BuildingImages> buildings{
+  {{Nation::Africans, BuildingType::Vineyard}, {1, 2, 3, 4, 5, 6, NOT_AVAILABLE, 7, 8}},
```

Comment:
Is it possible to keep the number of images constant, i.e. 9 images per building, 3 buildings per nation, 27 images per nation. That would make this redundant as a simple calculation can be used.

For LSTs there is the option to add an "empty" entry (in the file) which should work here.

Reply:
I can fix this. The only reason some of them have 9 instead of 8 is because some of the winterworld buildings need a separate shadow image for the building as a plant or sand missing means the shadow needs to be different.

Reply:
Sure I don't mind the empty ones. It is just about the bigish tables of pretty much only successive numbers in the code which could be a simple formula

Reply:
Would it be easier if I duplicate the shadow images that are unchanged for winterworld buildings so they can all be called for those buildings as the winterworld specific shadows are? Saving having to do an override for the few buildings that have a specific winter shadow, but rather all the winter shadows are called to the same multiple of 9?

Reply:
Otherwise I've just put an <empty> for buildings that don't have a winterworld shadow so each building is a group of 9 images.

Reply:
Done